src/main/java/book/store/repository/book/BookRepository.java

```diff
@@ -1,8 +1,31 @@
 package book.store.repository.book;
 
 import book.store.model.Book;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.data.jpa.repository.EntityGraph;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 public interface BookRepository extends JpaRepository<Book, Long>, JpaSpecificationExecutor<Book> {
+    @EntityGraph(attributePaths = "categories")
+    @Query("SELECT b FROM Book b JOIN FETCH b.categories c WHERE c.id = :id")
+    List<Book> findAllByCategoryId(@Param("id") Long id);

```

Comment:
Try to use Spring Data methods
This example should also work:
```suggestion
    List<Book> findAllByCategoriesId(Long id);
```